src/main/java/ch/njol/skript/effects/EffReplace.java

```diff
@@ -86,45 +91,50 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 		replacement = exprs[2 - matchedPattern % 2];
 		return true;
 	}
-	
-	@SuppressWarnings("null")
+
 	@Override
-	protected void execute(final Event e) {
+	@SuppressWarnings("null")
+	protected void execute(Event e) {
 		Object[] haystack = this.haystack.getAll(e);
 		Object[] needles = this.needles.getAll(e);
 		Object replacement = this.replacement.getSingle(e);
 		if (replacement == null || haystack == null || haystack.length == 0 || needles == null || needles.length == 0)
 			return;
-		if (replaceString) {
+		if (replaceString || replaceRegex) {
 			if (replaceFirst) {
 				for (int x = 0; x < haystack.length; x++)
-					for (final Object n : needles) {
+					for (Object n : needles) {
 						assert n != null;
-						haystack[x] = StringUtils.replaceFirst((String)haystack[x], (String)n, Matcher.quoteReplacement((String)replacement), caseSensitive);
+						haystack[x] = StringUtils.replaceFirst((String) haystack[x], (String) n, Matcher.quoteReplacement((String) replacement), caseSensitive);
+					}
+			} else if (replaceRegex) {
+				for (int x = 0; x < haystack.length; x++)
+					for (Object n : needles) {
+						assert n != null;
+						try {
+							haystack[x] = ((String) haystack[x]).replaceAll((String) n, (String) replacement);
+						} catch (Exception ignored) {}
 					}
 			} else {
 				for (int x = 0; x < haystack.length; x++)
-					for (final Object n : needles) {
+					for (Object n : needles) {
 						assert n != null;
 						haystack[x] = StringUtils.replace((String) haystack[x], (String) n, (String) replacement, caseSensitive);
 					}
 			}
 			this.haystack.change(e, haystack, ChangeMode.SET);
-		} else {
+		} else if (replaceItems) {
 			for (Inventory inv : (Inventory[]) haystack)
```

Comment:
Could you add curly brackets to these for statements ;D

(technically outside of this PR but you're editing the file)